java/registry/src/main/java/dev/sunbirdrc/registry/service/NativeSearchService.java

```diff
@@ -140,21 +141,27 @@ public JsonNode search(JsonNode inputQueryNode) throws IOException {
 
 		 	}
 		}
-		
+
 		return buildResultNode(searchQuery, result);
 	}
 
-	private void removeNonPublicFields(ArrayNode result, SearchQuery searchQuery, ObjectNode shardResult) throws Exception {
+	private ArrayNode removeNonPublicFields(SearchQuery searchQuery, ObjectNode shardResult) throws Exception {
+		ArrayNode result = JsonNodeFactory.instance.arrayNode();
 		for(String entityType: searchQuery.getEntityTypes()) {
 			ArrayNode arrayNode = (ArrayNode) shardResult.get(entityType);
-			for(JsonNode node : arrayNode) {
-				result.add(JSONUtil.removeNodesByPath(node, definitionsManager.getExcludingFieldsForEntity(entityType)));
+			if (removeNonPublicFieldsForNativeSearch) {
```

Comment:
by default, it should not remove non public fields while using GET API if the user is owner or schema is anonymous